node: add github packages support (#482)
CleverCloud/doc.clever-cloud.com: runtimes/nodejs/nodejs.md

You can (and should) define the `NODE_ENV=production` environment variable as many frameworks
rely on it for production mode.

### Special env NPM_TOKEN

> Since April 2015, npmjs.com allows you to have private repositories. The bad news is,

Reviewer: I wouldn't have deleted the whole bloc, keeping the explanation of why the customer may need those variables (private repositories)

> ### Use private repositories with NPM_REGISTRY and NPM_TOKEN

Reviewer: ```suggestion
### Use private repositories with CC_NPM_REGISTRY and NPM_TOKEN
```

> On Clever Cloud, you only need to provide the *token* part. So to register your auth token, you need 

Reviewer: Thats not really true anymore with CC_NPM_REGISTRY. I'd better see a text saying `To configure authentication on NPM, you can use NPM_TOKEN=token to configure the authentication token`. And then, `If you need to authenticate to another registry (github or a private one), you can also use CC_NPM_REGISTRY=<url> to configure the endpoint`

Something like that.

to add to your application the `NPM_TOKEN` environment variable.

``` javascript
Example:
NPM_TOKEN=00000000-0000-0000-0000-000000000000
```

Then, the .npmrc file will be created automatically for your application, with the registry url and the token.

``` javascript
//registry.npmjs.org/:_authToken=00000000-0000-0000-0000-000000000000
```

The default registry url is npmjs (registry.npmjs.org). We also support github packages (npm.pkg.github.com), you 

> can change the registry with the NPM_REGISTRY environment variable.

Reviewer 2: CC_NPM_REGISTRY

``` javascript
Example:

> NPM_REGISTRY=github

Reviewer 2: CC_NPM_REGISTRY

rely on it for production mode.

### Use private repositories with CC_NPM_REGISTRY and NPM_TOKEN

> Since April 2015, npmjs.com allows you to have private repositories. If you want to use a private repository on npmjs.com (the default one), you only need to provide the *token* part. So to register your auth token, you need to add to your application the `NPM_TOKEN` environment variable.

Reviewer: ```suggestion
Since April 2015, npmjs.com allows you to have private repositories. If you want to use a private repository on npmjs.com (the default one), you only need to provide the *token* part. To register your auth token, you need to add to your application the `NPM_TOKEN` environment variable.
```

//registry.npmjs.org/:_authToken=00000000-0000-0000-0000-000000000000
```

> To use another registry (like github), you can modify the url present in the .npmrc file with CC_NPM_REGISTRY the environment variable.

Reviewer: ```suggestion
To authenticate to another registry (like github), you can use the `CC_NPM_REGISTRY`  environment variable to define the registry's host.
```
